modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -361,7 +362,11 @@ public String extractIndexableAttributes(
 
 		Fields fields = _toFields(ddmStructure, ddmFormValues);
 
-		for (Field field : fields) {
+		for (DDMFormFieldValue ddmFormFieldValue :
+				ddmFormValues.getDDMFormFieldValues()) {
+
+			Field field = fields.get(ddmFormFieldValue.getName());
```

Comment:
@rolandpakai what about nested fields (`Field Group`)? As far as I know, `ddmFormValues.getDDMFormFieldValues()` won't return the nested fields.

Reply:
Hi @carolmariaabb 

indeed, we need to handle the Field Groups too.

Thanks, Roland